File: repos_estudiantes_ALVAREZ_primera_entrega/CARRERA_POZO_ALEJANDRA/IntroduccionPoo/Ejercicio7/Mascota.py
class Mascota:
    def __init__(self, nombre, tipo, energia):
        self.nombre = nombre
        self.tipo = tipo
        self.energia = energia

    def alimentar(self):
        """Aumenta la energía en +20, sin pasar de 100."""
        self.energia += 20
        if self.energia > 100:
            self.energia = 100
        print(f"{self.nombre} fue alimentado. Energía actual: {self.energia}")

    def jugar(self):
        """Disminuye la energía en -15, sin bajar de 0."""
        self.energia -= 15
        if self.energia < 0:
            self.energia = 0
        print(f"{self.nombre} jugó un rato. Energía actual: {self.energia}")

if __name__ == "__main__":
    mascota1 = Mascota("Luna", "Gato", 50)
    mascota2 = Mascota("Rocky", "Perro", 70)
    print(f"\nEnergía inicial de {mascota1.nombre}: {mascota1.energia}")
    print(f"Energía inicial de {mascota2.nombre}: {mascota2.energia}\n")
    mascota1.alimentar()
    mascota2.alimentar()
    mascota1.jugar()
    mascota2.jugar()

File: repos_estudiantes_ALVAREZ_primera_entrega/CHAMBI_BARRETO_CARLOS_ERNESTO/INTRODUCCION_A_POO/EJERCICIO7/Mascota.java
package INTRODUCCION_A_POO.EJERCICIO7;

public class Mascota {
    public String nombre, tipo;
    public int energia;

    public Mascota(String nombre, String tipo, int energia){
        this.nombre= nombre;
        this.tipo=tipo;
        this.energia=energia;
    }
    public void alimentar(){
        energia += 20;
        if(energia>100){
            energia=100;
        }
        System.out.println("La energia es: " + energia );
    }
    public void jugar(){
        energia -= 15;
        if(energia<0){
            energia=0;
        }
        System.out.println("Energia: " + energia);
    }
}

